app/src/components/button/button_stateful/index.tsx

```diff
@@ -40,29 +40,37 @@ const IdleDisabledCSS = css`
     opacity: 0.5;
   }
 `
+const SuccessCSS = css`
+  border: 1px solid ${props => props.theme.colors.green} !important;
+  color: ${props => props.theme.colors.green} !important;
+  opacity: 1 !important;
+`
 
 const Wrapper = styled.button<ButtonStatefulProps>`
-  ${ButtonCSS}
+  ${ButtonCSS};
   position: relative;
-  ${props => (props.state === ButtonStates.idle ? IdleDisabledCSS : DisabledCSS)};
+  ${props =>
+    props.state === ButtonStates.idle
+      ? IdleDisabledCSS
+      : props.state === ButtonStates.finished && props.extraText
+      ? SuccessCSS
+      : DisabledCSS};
 `
 
-const Text = styled.span<{ hide: boolean }>`
-  opacity: ${props => (props.hide ? '0' : '1')};
+const Text = styled.span<{ hide: boolean; state: ButtonStates }>`
+  display: ${props => (props.hide ? 'none' : 'flex')};
   position: relative;
   user-select: none;
   z-index: 1;
+  ${props => props.state === ButtonStates.idle && 'font-weight:500'};
```

Comment:
if you wanted to use the style library you could replace line 65 with 
`font-weight: ${props => props.state === ButtonStates.idle && props.theme.buttonSecondary.weight};`